include/exadg/incompressible_navier_stokes/postprocessor/line_plot_calculation_statistics.cpp

```diff
@@ -50,27 +50,33 @@ LinePlotCalculatorStatistics<dim, Number>::LinePlotCalculatorStatistics(
 
 template<int dim, typename Number>
 void
-LinePlotCalculatorStatistics<dim, Number>::setup(
-  LinePlotDataStatistics<dim> const & line_plot_data_in)
+LinePlotCalculatorStatistics<dim, Number>::setup(LinePlotData<dim> const & line_plot_data_in)
 {
   // initialize data
   data = line_plot_data_in;
 
-  if(data.statistics_data.calculate == true)
+  AssertThrow(
+    line_plot_data_in.time_control_data.trigger_every_sub_time_step !=
```

Comment:
In my opinion, `trigger_every_sub_time_step` should not be integrated into `TimeControl`. It does not have a meaning for most use cases of `TimeControl`. Also the name should be clearer, e.g. `write_to_file_every_time_steps`.

Reply:
An alternative is to introduce `TimeControlStatistics` (containing an additional variable) building upon `TimeControl` via inheritance or composition.

Reply:
This is fine with me

Reply:
composition is probably easier to realize because of `LinePlotData` and `LinePlotDataStatistics`.

Reply:
I think I got rid of ```LinePlotDataStatistics```

Reply:
Yes, but I can undo it if trigger_every_sub_time_step should live somewhere else

Reply:
I think we need

```c++
TimeControlStatistics
{

bool write_results()
{
  if(time_control.get_counter() % write_preliminary_results_every_nth_evaluation or reached_end_time())
  { return true; }
}

private:
TimeControl time_control;
double write_preliminary_results_every_nth_evaluation; 

};

Reply:
This version only requires to define another Data class, while the class `TimeControl` would be reused for statistics problems:

```c++
TimeControlStatisticsData
{
private:
TimeControlData time_control_data;
unsigned int write_preliminary_results_every_nth_evaluation; 

};

// not sure where to realize this exactly?
bool write_results()
{
  if(time_control.get_counter() % statistics_data.write_preliminary_results_every_nth_evaluation
     or time_control.reached_end_time())
  { do_write_results; }
}
```

Reply:
Sth like this? [b6a68c1](https://github.com/exadg/exadg/pull/262/commits/b6a68c149459d736ac598093701f85b89118f7a8)

Reply:
Yes, looks good overall. I am not sure if `write_preliminary_results()` is understandable from a very general `TimeControl` perspective, but we can still change this. This new solution has the advantage that it does not duplicate code.